Glasgow | Denis Suvorov | Form Controls | Week2

DenS116 opened the pull request:
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

DenS116 committed:
Added sceleton of form

DenS116 committed:
added data form

DenS116 committed:
Added styles to Html

DenS116 committed:
improved accessibility to 100 in Lighthouse and pinnned the footer tho the bottom of the page

DenS116 committed:
added focu-visible to CSS

DenS116 committed:
pin footer at the bottom and added styles to the button

DenS116 committed:
added tag label and added margin between radiobuttons

SallyMcGrath reviewed (commented):
Hi there @DenS116 . When you ran Lighthouse on this code, what did it tell you?

Try running it again and making the changes, like we did in class.

DenS116 commented:
> Всем привет@DenS116. Когда вы запустили Lighthouse для этого кода, что он вам сказал?
> 
> Попробуйте запустить его еще раз и внести изменения, как мы это делали на занятии.

Hi Sally, thankyou for your message and review my code. A
<img width="1680" alt="s" src="https://github.com/user-attachments/assets/1b83bbb4-d7d2-4e51-87b4-cbc2735cae30">
ccording to the requirements accessibility score should be 100 in lighthouse. I ran Lighthouse and my code meets the requirements

SallyMcGrath commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,120 @@
+        </div>
+
+        <div class="shop">
+          <p>Last Name:</p>
```
```suggestion
          <label for="last_name">Last Name:</label>
```

The label is labelling the input. In devtools, open the Accessibility tree and add some text to your empty label. Can you see the difference?



https://developer.chrome.com/docs/devtools/accessibility/reference#explore-tree

SallyMcGrath commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,120 @@
+
+        <h2>Choose color of the T-Shirt</h2>
+
+        <div class="shop shop_radio">
```
Instead of a div, is there a semantic HTML element you could use to group a set of fields?

SallyMcGrath reviewed (changes requested):
Great start @DenS116 . Take a look into what I'm saying about form elements. There's a good tutorial here https://webaim.org/techniques/forms/controls that might help

DenS116 committed:
filled label and replaced div with fieldset

DenS116 commented:
Hi @SallyMcGrath, I really appreciate for your feedback it really helped me to understand the "form" tag more deeply. I studied it in more detail and corrected the recommendations you gave me. I filled the "label" tag instead of 'p' and I changed the "div" tag to the "fieldset" tag which it is schematically more suitable for the form. I will be glad to hear recommendations, thanks
